pkg/diagnose/k8s_version.go

```diff
@@ -44,7 +46,13 @@ func K8sVersion(client kubernetes.Interface, status reporter.Interface) bool {
 		return false
 	}
 
-	status.Success("Kubernetes version %q is supported", k8sVersion)
+	brokerSecret, err := secret.EnsureFromSA(client, constants.SubmarinerBrokerAdminSA, "submariner-k8s-broker")
+	if err != nil {
+		status.Failure(err.Error())
+		return false
+	}
```

Comment:
This function checks K8s version support so how is this code related? Also the "submariner-k8s-broker" namespace is hard-coded which is problematic.

Reply:
> This function checks K8s version support so how is this code related? Also the "submariner-k8s-broker" namespace is hard-coded which is problematic.

This is not for fix, this was just an easy way to test the changes without having to fully deploy submariner while working on code changes.